evalml/pipelines/components/transformers/preprocessing/time_series_featurizer.py

```diff
@@ -234,33 +253,28 @@ def _compute_delays(self, X_ww, y):
             ).columns,
         )
         categorical_columns = self._get_categorical_columns(X_ww)
-        cols_derived_from_categoricals = []
         lagged_features = {}
         if self.delay_features and len(X_ww) > 0:
             X_categorical = self._encode_X_while_preserving_index(
                 X_ww[categorical_columns],
             )
-            for col_name in cols_to_delay:
-                col = X_ww[col_name]
-                if col_name in categorical_columns:
-                    col = X_categorical[col_name]
-                for t in self.statistically_significant_lags:
-                    feature_name = f"{col_name}_delay_{t}"
-                    lagged_features[f"{col_name}_delay_{t}"] = col.shift(t)
-                    if col_name in categorical_columns:
-                        cols_derived_from_categoricals.append(feature_name)
+            lagged_features.update(
+                self._delay_df(X_ww, cols_to_delay, categorical_columns, X_categorical),
+            )
         # Handle cases where the target was passed in
         if self.delay_target and y is not None:
-            if type(y.ww.logical_type) == logical_types.Categorical:
-                y = self._encode_y_while_preserving_index(y)
-            for t in self.statistically_significant_lags:
-                lagged_features[self.target_colname_prefix.format(t)] = y.shift(t)
+            if isinstance(y, pd.DataFrame):
+                lagged_features.update(self._delay_df(y, y.columns))
```

Comment:
thought: should we just run `self._encode_y_while_preserving_index(y)` even though we won't expect categorical columns just yet?

Reply:
🤔 interesting point, will we ever expect categorical columns? We're only supporting regression problems for multiseries

Reply:
potentially sometime in the distant future! Just thought it would make it one step easier for whoever implements that 😄  it'll be a no-op anyways right now

Reply:
I like the idea in theory, but I'm worried about increasing runtime with checking if `y` contains any categorical columns. We'd have to do so in all cases, which feels wasteful when we know we won't be dealing with it.